MariaDB/server
# MDEV-27882 Innodb - recognise MySQL-8.0 innodb flags and give a specific error message

- [x] *The Jira issue number for this PR is: MDEV-MDEV-27882*


## Description

Per fsp0types.h, SDI is on tablespace flags position 14 where MariaDB stores its pagesize. fsp_flags_is_valid has already returned an error for this "ssize == 2" corresponding to this flag.

So we assume its MySQL-8.0 because of this SDI flag and give the user a helpful message.

## How can this PR be tested?

Using a mysql-8.0 datadir in /tmp/m:
```
$ sql/mysqld --no-defaults --skip-networking --datadir=/tmp/m --socket=/tmp/ssock  --verbose
2022-11-07 16:09:50 0 [Note] sql/mysqld (mysqld 10.3.37-MariaDB) starting as process 175615 ...
2022-11-07 16:09:50 0 [Note] InnoDB: Using Linux native AIO
2022-11-07 16:09:50 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2022-11-07 16:09:50 0 [Note] InnoDB: Uses event mutexes
2022-11-07 16:09:50 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-11-07 16:09:50 0 [Note] InnoDB: Number of pools: 1
2022-11-07 16:09:50 0 [Note] InnoDB: Using SSE2 crc32 instructions
2022-11-07 16:09:51 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
2022-11-07 16:09:51 0 [Note] InnoDB: Completed initialization of buffer pool
2022-11-07 16:09:51 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2022-11-07 16:09:51 0 [ERROR] InnoDB: MySQL-8.0 tablespace in ./ibdata1 detected. Restart in MySQL-8.0 for migration/recovery.
2022-11-07 16:09:51 0 [ERROR] InnoDB: Plugin initialization aborted with error Unsupported
2022-11-07 16:09:51 0 [Note] InnoDB: Starting shutdown...
2022-11-07 16:09:51 0 [ERROR] Plugin 'InnoDB' init function returned error.
2022-11-07 16:09:51 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
2022-11-07 16:09:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2022-11-07 16:09:51 0 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2022-11-07 16:09:51 0 [ERROR] Unknown/unsupported storage engine: InnoDB
2022-11-07 16:09:51 0 [ERROR] Aborting
```


## Basing the PR against the correct MariaDB version


## Backward compatibility
TODO: fill details here, if applicable, or remove the section
